### PR TITLE
Doc/tutorial fix

### PR DESCRIPTION
**Pull request by FranFin**

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/soss-dds-linux/20/)

File: doc/tutorial.md [hunk 1]
```diff
@@ -1,33 +1,33 @@
...
 
 To install it, you can take a look to the
 [official documentation](https://index.ros.org/doc/ros2/Installation/Eloquent/Linux-Install-Binary/).
 
-In order to test it(supposing eloquent), please open two terminals and do the following:
+In order to test it (supposing eloquent), please open two terminals and do the following:
 - First terminal:
   ```
-  source ~/ros2_eloquent/ros2-linux/setup.bash
+  source ~/ros2_eloquent/install/setup.bash
```
Comment (lemunozm) [inline doc/tutorial.md]: This seems to be so, but the previously linked page shows that path (https://index.ros.org/doc/ros2/Installation/Eloquent/Linux-Install-Binary/#sourcing-the-setup-script).

Maybe the tutorial is bad, did you check it?

File: doc/tutorial.md [hunk 2]
```diff
@@ -1,33 +1,33 @@
...
   ros2 run demo_nodes_cpp talker
   ```
 - Second terminal:
   ```
-  source ~/ros2_eloquent/ros2-linux/setup.bash
+  source ~/ros2_eloquent/install/setup.bash
```
Comment (lemunozm) [inline doc/tutorial.md]: Same comment here

File: doc/tutorial.md [hunk 3]
```diff
@@ -37,43 +37,43 @@ Note that `Fast-RTPS-Gen` is a Java application. You need to have installed `Jav
...
+Four our purpose, we need to add two repositories to our workspace:
 - `https://github.com/osrf/soss` that offers the soss core and ros2 system handle.
 - `https://github.com/eProsima/soss-dds` that offers the dds system handle.
 Both repositories must be placed into the `src` folder of our _colcon workspace_.
 
-Before compile it with colcon, we need to _source_ the ros2 environment as follows (supposing eloquent and linux):
+Before compiling it with colcon, we need to _source_ the ros2 environment as follows (supposing eloquent and linux):
 ```
-source ~/ros2_eloquent/ros2-linux/setup.bash
+source ~/ros2_eloquent/install/setup.bash
```
Comment (lemunozm) [inline doc/tutorial.md]: Same comment here

File: doc/tutorial.md [hunk 4]
```diff
@@ -37,43 +37,43 @@ Note that `Fast-RTPS-Gen` is a Java application. You need to have installed `Jav
...
-Then we can compile simply writing the following command from our colcon workspace:
+Then we can compile by simply writing the following command from our colcon workspace:
 ```
 colcon build
 ```
 
 ## DDS-ROS2 use case
 In our use case we want communicate a ros2 world that is using the `std_msgs/String` type,
-with a dds world that is using the `HelloWorld.idl` found in the _fastrtps getting started_.
+with a dds world that is using a dds type defined in the `HelloWorld.idl` file found in the _fastrtps getting started_.
```
Comment (lemunozm) [inline doc/tutorial.md]:
```suggestion
with a dds world that is using a type defined in the `HelloWorld.idl` file found in the _fastrtps getting started_.
```
dds seems redundant here

File: doc/tutorial.md [hunk 5]
```diff
@@ -96,19 +96,19 @@ topics:
...
-To archive this, we need to modify the topic name of both, publisher and subscriber `.cxx` example files.
+The previously generated example of Fast-RTPS communicates the dds world publisher and subscriber between them.
+Here we do not want this behavior, and we want instead to publish from `dds` into `ros2` and to subscribe to `dds` from `ros2`.
+To achieve this, we need to modify the topic names of both the publisher and subscriber `.cxx` example files.
 
 #### `HelloWorldPublisher.cxx`
 Search the line containing the `topicName` of the publisher
 ```c++
-Wparam.topic.topicName = "Vector2iPub";
+Wparam.topic.topicName = "HelloWorldPubSubTopic";
```
Comment (lemunozm) [inline doc/tutorial.md:109]: Upss!! You are right!

File: doc/tutorial.md [hunk 6]
```diff
@@ -96,19 +96,19 @@ topics:
...
 ```
-and modify with the following topic Name:
+and modify it with a different topic name, for instance the following:
```
Comment (lemunozm) [inline doc/tutorial.md]:
```suggestion
and modify it with a different topic name, as follows:
```
For the proposed `yaml`, this change is mandatory to make it works

File: doc/tutorial.md [hunk 7]
```diff
@@ -125,62 +125,60 @@ st.msg("This is a message from dds to ros2");
...
 
 #### `HelloWorldSubscriber.cxx`
-Analogous to the first publisher modification, search the `topicName` line
+Analogously to the above publisher modification, search the `topicName` line of the subscriber
 ```c++
-Wparam.topic.topicName = "Vector2iSub";
+Wparam.topic.topicName = "HelloWorldPubSubTopic";
 ```
-and modify with the following topic Name:
+and modify it with a different topic name, for instance the following:
```
Comment (lemunozm) [inline doc/tutorial.md]:
```suggestion
and modify it with a different topic name, as follows:
```

File: doc/tutorial.md [hunk 8]
```diff
@@ -125,62 +125,60 @@ st.msg("This is a message from dds to ros2");
...
 ```c++
 Wparam.topic.topicName = "hello_dds";
 ```
 
 Once the files have been modified, we need to compile them.
 Simply, write `make` in your build folder.
 
-If we run now the publisher and subscriber, they will not match among them.
+If we now run the dds world publisher and subscriber, they will not match between them.
```
Comment (lemunozm) [inline doc/tutorial.md]:
```suggestion
If we now run both publisher and subscriber, they will not match between them.
```
This subsection is only talking about `dds`

File: doc/tutorial.md [hunk 9]
```diff
@@ -125,62 +125,60 @@ st.msg("This is a message from dds to ros2");
...
+To deploy the system we will need 5 terminals at the same time:
 ![](images/terminal_deploy.png)
 
 #### Terminal 1
-To publish the topic from ros2, only need to write the following commands:
+To publish the topic from ros2, we only need to write the following commands:
 ```
-source ~/ros2_eloquent/ros2-linux/setup.bash
-ros2 topic pub /hello_dds std_msgs/String "This is a message from ros2 to dds"
+source ~/ros2_eloquent/install/setup.bash
```
Comment (lemunozm) [inline doc/tutorial.md]: Same commet as before

File: doc/tutorial.md [hunk 10]
```diff
@@ -125,62 +125,60 @@ st.msg("This is a message from dds to ros2");
...
+ros2 topic pub /hello_dds std_msgs/String "{\"This is a message from ros2 to dds\"}"
 ```
 
 #### Terminal 2
-Similar to publish from ros2, for subscribe we need to write:
+Similarly to what we did to publish from ros2, for subscribing we need to write:
 ```
-source ~/ros2_eloquent/ros2-linux/setup.bash
+source ~/ros2_eloquent/install/setup.bash
```
Comment (lemunozm) [inline doc/tutorial.md]: Same commet as before

File: doc/tutorial.md [hunk 11]
```diff
@@ -125,62 +125,60 @@ st.msg("This is a message from dds to ros2");
...
 ros2 topic echo /hello_ros std_msgs/String
 ```
 
 #### Terminal 3
-Placed into the build folder when you have generated the HelloWorld publisher and subscriber.
-Run the publisher
+Placed into the build folder where you have generated the HelloWorld publisher and subscriber, run the dds world publisher
```
Comment (lemunozm) [inline doc/tutorial.md]:
```suggestion
Placed into the build folder where you have generated the HelloWorld publisher and subscriber, run the dds publisher
```
The "jargon" when you refer to it is `dds publisher` without `world`

File: doc/tutorial.md [hunk 12]
```diff
@@ -125,62 +125,60 @@ st.msg("This is a message from dds to ros2");
...
 ```
 ./HelloWorld publisher
 ```
 
 #### Terminal 4
-Similar to terminal 4, but running the subscriber:
+As in terminal 3, run the dds world subscriber:
```
Comment (lemunozm) [inline doc/tutorial.md]:
```suggestion
As in terminal 3, run the dds subscriber:
```

File: doc/tutorial.md [hunk 13]
```diff
@@ -125,62 +125,60 @@ st.msg("This is a message from dds to ros2");
...
 ```
 ./HelloWorld subscriber
 ```
 
 #### Terminal 5
-Before running soss, any matched is made.
-Placed into the `soss` workspace.
-Source both, ros2 and your installation, and run the `soss` executable with the previous yaml.
+All matchings are made before running soss.
```
Comment (lemunozm) [inline doc/tutorial.md]: This sentence means the opposite, maybe I have wrongly expressed my self. I want to say that no matching has be done before soss

File: doc/tutorial.md [hunk 14]
```diff
@@ -125,62 +125,60 @@ st.msg("This is a message from dds to ros2");
...
+Placed into the `soss` workspace, source both ros2 and your soss installation, and run the `soss` executable with the previous yaml.
 
 ```
-source ~/ros2_eloquent/ros2-linux/setup.bash
+source ~/ros2_eloquent/install/setup.bash
```
Comment (lemunozm) [inline doc/tutorial.md]: Same as before

**Review by lemunozm — CHANGES_REQUESTED** [review]

File: doc/tutorial.md (hunk 13 above)
Comment (FranFin) [reply doc/tutorial.md]: Ah ok. So maybe we could change it to: "No matching is made before running soss.".

File: doc/tutorial.md (hunk 1 above)
Comment (FranFin) [reply doc/tutorial.md]: Yes I checked it. It doesn't work for me. On my pc it only works with "install" in spite of "ros2-linux", that's why I changed it.

File: doc/tutorial.md (hunk 2 above)
Comment (FranFin) [reply doc/tutorial.md]: Same.

File: doc/tutorial.md (hunk 3 above)
Comment (FranFin) [reply doc/tutorial.md]: Same.

File: doc/tutorial.md (hunk 14 above)
Comment (FranFin) [reply doc/tutorial.md]: Same.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/soss-dds-linux/22/)

File: doc/tutorial.md (hunk 1 above)
Comment (lemunozm) [reply doc/tutorial.md]: As we talk, this is due to the installation way, in this case, by binaries.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/soss-dds-linux/24/)

File: doc/tutorial.md (hunk 13 above)
Comment (lemunozm) [reply doc/tutorial.md]: Ok perfect!

**Review by lemunozm — APPROVED** [review]

**Pull request merged**